java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageReader.java

```diff
@@ -18,48 +18,50 @@
 
 package org.apache.arrow.vector.ipc.message;
 
-
-import io.netty.buffer.ArrowBuf;
-import org.apache.arrow.flatbuf.Message;
-import org.apache.arrow.memory.BufferAllocator;
-
 import java.io.IOException;
 
 /**
- * Interface for reading a sequence of messages.
+ * Abstract base class for reading a sequence of messages.
  */
-public interface MessageReader {
+public abstract class MessageReader<T extends MessageReadHolder> {
```

Comment:
Please make this AutoCloseable with the close throwing the generic parameter suggested above for exception.

Reply:
Sure, I can do that